src/subdomains/supporting/payin/services/payin.service.ts

```diff
@@ -96,19 +94,27 @@ export class PayInService {
     await this.payInRepository.save(payIn);
   }
 
-  async updateAmlCheck(payInId: number, amlCheck: CheckStatus): Promise<void> {
-    await this.payInRepository.update(payInId, { amlCheck });
+  async updateForwardConfirmation(payInId: number, forwardConfirmation: boolean): Promise<void> {
+    await this.payInRepository.update(payInId, { isForwardConfirmed: forwardConfirmation });
   }
 
   async returnPayIn(
     payIn: CryptoInput,
     purpose: PayInPurpose,
-    returnAddress: BlockchainAddress,
+    returnAddress: string,
     route: Staking | Sell | Swap,
+    chargebackAmount: number,
   ): Promise<void> {
-    const amlCheck = route.user.userData.kycLevel === KycLevel.REJECTED ? CheckStatus.FAIL : CheckStatus.PASS;
-    payIn.triggerReturn(purpose, returnAddress, route, amlCheck);
+    if (payIn.isForwardConfirmed) throw new BadRequestException('CryptoInput already forwarded');
 
+    payIn.triggerReturn(
+      purpose,
+      BlockchainAddress.create(returnAddress, payIn.asset.blockchain),
+      route,
+      chargebackAmount,
+    );
+
+    // Remove that?
     if (payIn.transaction)
```

Comment:
TODO?

Reply:
It's more a question than a todo 😁 
We should decide if it makes sense or not

Reply:
Yes, remove, if Robin agrees